### PR TITLE
Bump `kv-asset-handler` to local workspace

### PR DESCRIPTION
**Pull request by itsmatteomanf**
**What this PR solves / how to test:**

Bumps `kv-asset-handler` to the latest v0.3.0 release, no breaking changes, just fixes.

**Author has included the following, where applicable:**

- [ ] Tests
- [x] Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))

**Reviewer is to perform the following, as applicable:**

- Checked for inclusion of relevant tests
- Checked for inclusion of a relevant changeset
- Checked for creation of associated docs updates
- Manually pulled down the changes and spot-tested

### CODE REVIEW

**Comment by petebacondarwin** [comment]
I believe we achieved this when we brought the kv-asset-handler over to the workers-sdk monorepo.

**Comment by itsmatteomanf** [comment]
@petebacondarwin the files look unchanged to me in the current main branch...

**Comment by petebacondarwin** [comment]
OK if you could rebase on main then we can see the diff.

**Review by petebacondarwin — APPROVED** [review]
Assuming the CI agrees.

File: .changeset/sharp-snails-obey.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"wrangler": patch
```
Comment (petebacondarwin) [inline .changeset/sharp-snails-obey.md:2]: Strictly this is not a change to the Wrangler package. It is only changing some fixtures and templates, which are not deployed as part of Wrangler.
I think we could actually dispense with this changeset altogether.

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
One last thing, regarding the changeset.
But otherwise looks smashing! Thanks.

**Comment by itsmatteomanf** [comment]
I updated it so that it uses the `workspace:*` syntax in all cases...

**Review by petebacondarwin — APPROVED** [review]

**Pull request merged**